joomla/joomla-cms
# [4.0] Show a list of events a plugin subscribes to in its edit page if plugin implements SubscriberInterface

### Summary of Changes

At the moment there is really no way to know what events a plugin is subscribed to without reading the code.  The 3.x and earlier plugin convention arbitrarily assumes that all public methods in a plugin class are potentially event listeners and are treated as such.  With the Framework's dispatcher in use in 4.0 and the beginning of the transition toward plugins which use a "configuration over convention" style approach, we can start finding this information and use it as practical.

So, for 4.0, for plugins which implement `Joomla\Event\SubscriberInterface` we can add a new tab to the plugin's edit page showing the list of events that are subscribed to.  This PR accomplishes just that.

### Testing Instructions

With the patch applied, go to the Plugin Manager and edit the "Quick Icon - Joomla Update Notification" plugin.  You should see a new tab on this page as the plugin has been updated to implement the above referenced interface.  When editing any other plugin, the tab won't be there as no other plugin has been updated.

### Expected result

<img width="1440" alt="screen shot 2018-02-05 at 7 40 24 pm" src="https://user-images.githubusercontent.com/368545/35837568-7b144c92-0aac-11e8-9105-3bae7c302070.png">

### Documentation Changes Required

N/A, plugin manager practices aren't documented anywhere.
